src/Chapter2.hs

```diff
@@ -630,7 +630,8 @@ takeEven = go []
   where
     go :: [Int] -> [Int] -> [Int]
     go acc [] = acc
-    go acc (x:xs) = if even x then go (acc ++ [x]) xs else go acc xs
+    go acc [x] = x : acc
+    go acc (x : _ : xs) = go (x : acc) xs
```

Comment:
Unfortunately, by adding elements from the beginning of the argument to the beginning of the result, you get elements in the reversed order (similar to the `rewind` implementation). You need to add the beginning of the result to get in the same order. After that, you can also notice, that the variable `acc` becomes redundant :slightly_smiling_face: 
```suggestion
    go acc (x : _ : xs) = x : go acc xs
```